example/fetchMock.ts

```diff
@@ -1,16 +1,55 @@
 import fetchMock from 'fetch-mock';
-import FakeRest from 'fakerest';
+import { FetchMockServer, withDelay } from '../src';
 import { data } from './data';
+import { dataProvider as defaultDataProvider } from './dataProvider';
 
 export const initializeFetchMock = () => {
-    const restServer = new FakeRest.FetchServer({
+    const restServer = new FetchMockServer({
         baseUrl: 'http://localhost:3000',
+        data,
+        loggingEnabled: true,
     });
     if (window) {
         // @ts-ignore
         window.restServer = restServer; // give way to update data in the console
     }
-    restServer.init(data);
-    restServer.toggleLogging(); // logging is off by default, enable it
+
+    restServer.addMiddleware(withDelay(300));
+    restServer.addMiddleware(async (request, context, next) => {
+        if (!request.headers?.get('Authorization')) {
+            return {
+                status: 401,
+                headers: {},
+            };
+        }
+        return next(request, context);
+    });
+    restServer.addMiddleware(async (request, context, next) => {
+        if (context.collection === 'books' && request.method === 'POST') {
+            if (
+                restServer.collections[context.collection].getCount({
+                    filter: {
+                        title: context.requestBody?.title,
+                    },
+                }) > 0
+            ) {
+                throw new Response(
+                    JSON.stringify({
+                        errors: {
+                            title: 'An article with this title already exists. The title must be unique.',
+                        },
+                    }),
+                    {
+                        status: 400,
+                        statusText: 'Title is required',
+                    },
+                );
```

Comment:
I still use a throw Response here instead of returning an object to test that this still works